From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Wed, 5 Jun 2024 20:57:09 -0500
Subject: [PATCH] Always initialize econf_file

As of libeconf 0.7.0 econf_file objects must be initialized before use.

---
 modules/pam_shells/pam_shells.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/pam_shells/pam_shells.c b/modules/pam_shells/pam_shells.c
index d5f8ec35..cf1e5390 100644
--- a/modules/pam_shells/pam_shells.c
+++ b/modules/pam_shells/pam_shells.c
@@ -81,7 +81,7 @@ static int perform_check(pam_handle_t *pamh)
     size_t size = 0;
     econf_err error;
     char **keys;
-    econf_file *key_file;
+    econf_file *key_file = NULL;
 
     error = econf_readDirsWithCallback(&key_file,
 				       VENDORDIR,
